src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource.cs

```diff
@@ -162,15 +184,11 @@ public bool TryGetTelemetryId(out Guid telemetryId)
 
                 if (_workspaceStatusService != null)
                 {
-                    // TODO: right now, LightBulb uses IVsThreadedWaitDialog directly rather than new
-                    //       IUIThreadOperationContext. we need to talk to editor team whether we want to
-                    //       update API to pass in OperationContext like any new API, or we use 
-                    //       IWaitIndicator abstraction (which is a thin wrapper on top of IVsThreadedWaitDialog) directly
-                    //       for now, we use the one LB created before calling us. meaning we don't update
-                    //       text on the wait dialog window
-                    // 
-                    //       this also needs to run under threading context otherwise, we can deadlock on VS
-                    ThreadingContext.JoinableTaskFactory.Run(() => _workspaceStatusService.WaitUntilFullyLoadedAsync(cancellationToken));
+                    using (addOperationScope("WorkspaceStatusService"))
```

Comment:
this string isn't user visible right?

Reply:
It is user visible, do you have a better name suggestion? Note that this is shown as `Gathering Suggestions - '{0}'`, so it is not a localized string, but expected to be the name of the component that is executing.

Reply:
I think users will be confused by this message if it shows (i.e. what is 'WorkspaceStatusService'). We should check with PMs about better message for this case.

Reply:
`Gathering Suggestions - Waiting for the solution to fully load`.

Reply:
Yeah that works for me. We should keep the `'` characters or use a separate resource string though.

Reply:
I'll use a separate resource string.